engine/server/engine/main.go

```diff
@@ -131,6 +134,40 @@ func runMain() error {
 		},
 	)
 
+	go func() {
+		staticPath := "/run/webapp"
+		indexPath := "index.html"
+
+		fileServer := http.FileServer(http.Dir(staticPath))
+
+		handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			path, err := filepath.Abs(r.URL.Path)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				return
+			}
+
+			path = filepath.Join(staticPath, path)
+
+			_, err = os.Stat(path)
+			if os.IsNotExist(err) {
+				// file does not exist, serve index.html
+				http.ServeFile(w, r, filepath.Join(staticPath, indexPath))
+				return
+			} else if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			fileServer.ServeHTTP(w, r)
+		})
+
+		err := http.ListenAndServe(webappPortAddr, handler)
+		if err != nil {
+			logrus.Debugf("error while starting the webapp: \n%v", err)
+		}
```

Comment:
you could even use a defer call I think, like `defer server.Shutdown(ctx)`

Reply:
sorry; why do we need this though? Because kurtosis engine is running inside docker container; so we always stop/delete old and and start new one? so it will be abruptly close anyways? @leoporoli maybe I am missing something here?

Reply:
@Peeeekay stopping the server gracefully will return the received HTTP request before the engine stops

Reply:
@leoporoli Ah I see; I don't think it is absolutely necessary to do that frontend - maybe overkill here - because as soon as an engine gets up - it should work fine. I don;t really want to impact the current behaviour which is kill the engine as soon as  grpcServer is gracefully stopped. Though this is one-liner change and can be done later on.